Porting asynchronous example from NAN repo

NickNaso opened the pull request:
The original example is hosted here:
https://github.com/nodejs/nan/tree/master/examples/async_pi_estimate
Rewrite all the logic using **node-addon-api**
See the following issue
https://github.com/nodejs/abi-stable-node-addon-examples/issues/10

NickNaso committed:
Porting asynchronous example from NAN repo
The original example is hosted here:
https://github.com/nodejs/nan/tree/master/examples/async_pi_estimate
Rewrite all the logic using **node-addon-api**
See the following issue
https://github.com/nodejs/abi-stable-node-addon-examples/issues/10

mhdawson commented on async_pi_estimate/node-addon-api/async.cc:
```diff
@@ -0,0 +1,39 @@
+
+  // Executed when the async work is complete
+  // this function will be run inside the main event loop
+  // so it is safe to use V8 again
```
probably don't want the reference to V8 here.

NickNaso replied:
Replaced V8 with JS engine

mhdawson commented on async_pi_estimate/node-addon-api/async.cc:
```diff
@@ -0,0 +1,39 @@
+  // this function will be run inside the main event loop
+  // so it is safe to use V8 again
+  void OnOK() {
+     Napi::HandleScope scope(Env());   
```
extra space ?

NickNaso replied:
Removed space

mhdawson commented on async_pi_estimate/node-addon-api/pi_est.h:
```diff
@@ -0,0 +1,14 @@
+ * Copyright (c) 2018 NAN contributors
+ *
+ * MIT License <https://github.com/nodejs/nan/blob/master/LICENSE.md>
+ ********************************************************************/
```
I don't see this copyright in any of the files being added under nan. I'd expect it to be in both or neither.

NickNaso replied:
Removed copyright

NickNaso committed:
Removed copyright info

NickNaso committed:
Removed extra spacxe

NickNaso committed:
Replaced V8 with JS engine

mhdawson reviewed (approved):
LGTM

NickNaso committed:
Added examples to use event emitter on native addon-ons

NickNaso committed:
Deleted bad examples

NickNaso committed:
Added examples on how to use event emitter ref. https://github.com/nodejs/node-addon-api/issues/110

NickNaso committed:
Removed bad examples

NickNaso committed:
Added examples for event emitter
See: https://github.com/nodejs/node-addon-api/issues/110

NickNaso committed:
Added examples to use event emitter on native addon-ons

Deleted bad examples


Added examples on how to use event emitter ref. https://github.com/nodejs/node-addon-api/issues/110

Removed bad examples

Added examples for event emitter
See: https://github.com/nodejs/node-addon-api/issues/110

NickNaso committed:
Merge remote-tracking branch 'origin/master'

NickNaso commented:
Hi @mhdawson I added two examples about event emitter. Could you review the code? Thanks!

mhdawson reviewed (approved):
LGTM

mhdawson commented:
Landed as b6f3e8beed478baac296bb224fa9902d6abad119
